//python -m grpc_tools.protoc --proto_path=. ./services.proto --python_out=. --grpc_python_out=.
syntax = "proto3";
package services;

import "google/protobuf/empty.proto";

// The Information service definition.
service Information {
    /*
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    // Sends a file
    rpc Chunker(google.protobuf.Empty) returns (stream Chunk) {}*/

    rpc SendCounter (google.protobuf.Empty) returns (CounterReply) {}
    rpc SendBooks (google.protobuf.Empty) returns (BooksReply) {}
    rpc SendDB (google.protobuf.Empty) returns (stream File) {}
    rpc SendHour (google.protobuf.Empty) returns (HourReply) {}
    rpc SendBook (google.protobuf.Empty) returns (BookReply) {}
  }

//Responses for other servers
message CounterReply{
  int32 counter = 1;
}

message BooksReply{
  string books = 1;
}

message File{
  bytes chunk = 1;
}

//Responses for clients
message HourReply{
    string hour = 1;
}

message BookReply{
  string book = 1;
}



/*
  // The request message containing the user's name.
  message HelloRequest {
    string name = 1;
  }

  // The response message containing the greetings
  message HelloReply {
    string message = 1;
  }

  message Chunk {
    bytes chunk = 1;
  }
*/